projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
+
+The alternative version with an array of ids goes like this:
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities([
+    { id: 2, name: 'Car Washing', finished: false },
+    { id: 3, name: 'Cat Feeding', finished: false },
+    { id: 4, name: 'Dog Feeding', finished: false },
+  ])
+);
+
+patchState(
+  this.todoStore,
+  updateEntities({
+    ids: [2, 3],
+    changes: { finished: true },
+  })
+);
+
+patchState(this.todoStore, removeEntities([2, 4]));
+```
+
+## Customized Id property
+
+The default property name for an identifier is `id` and is of type `string` or `number`.
+
+It is possible to specify a different name, but the type must still be `string` or `number`. You can specify the id only when adding or setting an entity. It is not possible to define it via `withEntities`.
+
+Therefore, all variations of the `add*` and `set*` functions have an optional (last) parameter, which is an object literal and allows to define the id property via `idKey`.
+
+For example:
+
+```typescript
+interface Todo {
+  key: number;
+  name: string;
+  finished: boolean;
+}
+
+patchState(
+  this.todoStore,
+  addEntities(
+    [
+      { key: 2, name: 'Car Washing', finished: false },
+      { key: 3, name: 'Cat Feeding', finished: false },
+    ],
+    { idKey: 'key' }
+  )
+);
+
+patchState(this.todoStore, setEntity({ key: 4, name: 'Dog Feeding', finished: false }, { idKey: 'key' }));
+```
+
+The `update*` and `remove*` methods, which expect an id value, will automatically pick the right one. That is possible because every entity belongs to a map with its id as the key.
+
+Theoretically, adding the same entity twice with different id names would be possible. For obvious reasons, we discourage you from doing that.
+
+## Multiple Entity Types / Named Entities
+
+If you don't want to have the property names `ids`, `entities` and `entityMap` in the state, you can define others by providing a collection name.
+
+Another use case is to have multiple entity types in one store. For example, `Todo` and `User`. In that case, you also have to come up with collections.
+
+The following example shows a store that already has an `ids` property in its state before applying `withEntities`. Therefore, we provide the collection name `todo`.
+
+```typescript
+interface Todo {
+  id: number;
+  name: string;
+  finished: boolean;
+}
+
+const TodoStore = signalStore(
+  withState({ ids: [] }), // ids property already exists
+  withEntities({ entity: type<Todo>(), collection: 'todo' })
+);
+
+@Component({
+  selector: 'ngrx-entity',
+  template: `@for (todo of todoStore.todoEntities(); track todo.id) {
+    <p>Current Ids: {{ todoStore.todoIds() }}</p>
+    <ul>
+      <li [ngStyle]="{ 'font-weight': todo.finished ? 'bold' : 'normal' }">
+        {{ todo.name }}
+      </li>
+    </ul>
+
+    }`,
+  standalone: true,
+  providers: [TodoStore],
+  imports: [NgStyle],
+})
+export class EntityComponent implements OnInit {
```

Comment:
Check the suggestion for the previous component - `<ul>` should be out of `@for`, component name can be `TodosComponent`, selector `ngrx-todos`, etc.

Reply:
Done